fed/mosaic-mapping/src/main/java/org/eclipse/mosaic/fed/mapping/ambassador/MappingAmbassador.java

```diff
@@ -131,28 +133,28 @@ private void handleInteraction(ScenarioVehicleRegistration scenarioVehicle) thro
                 );
                 return;
             }
-
+            List<String> applications = prototype.applications == null ? new ArrayList<>() : prototype.applications;
             if (randomNumberGenerator.nextDouble() >= ObjectUtils.defaultIfNull(prototype.weight, 1.0)) {
                 log.debug(
                         "This scenario vehicle \"{}\" of prototype \"{}\" will not be equipped due to a weight condition of {}.",
                         scenarioVehicle.getVehicleTypeId(),
                         scenarioVehicle.getId(),
                         prototype.weight
                 );
-                return;
+                applications = new ArrayList<>();
             }
 
             final VehicleRegistration vehicleRegistration = new VehicleRegistration(
                     scenarioVehicle.getTime(),
                     scenarioVehicle.getName(),
                     prototype.group,
```

Comment:
I would think that the group should also only be used from the prototype on the same conditions as the list of applications. In the case the weight condition is false, we should pass `null` instead. (?)